packages/formdescriptioneditors/backend/sirius-components-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/formdescriptioneditors/components/PieChartStyleProvider.java

```diff
@@ -0,0 +1,88 @@
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.formdescriptioneditors.components;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.eclipse.sirius.components.charts.piechart.components.PieChartStyle;
+import org.eclipse.sirius.components.charts.piechart.components.PieChartStyle.Builder;
+import org.eclipse.sirius.components.view.PieChartDescriptionStyle;
+
+/**
+ * The style provider for the PieChart Description widget of the View DSL.
+ *
+ * @author arichard
+ */
+public class PieChartStyleProvider {
+
+    private final PieChartDescriptionStyle viewPieChartDescriptionStyle;
+
+    public PieChartStyleProvider(PieChartDescriptionStyle viewPieChartDescriptionStyle) {
+        this.viewPieChartDescriptionStyle = Objects.requireNonNull(viewPieChartDescriptionStyle);
+    }
+
+    public PieChartStyle build() {
+        Builder pieChartStyleBuilder = PieChartStyle.newPieChartStyle();
+        String colors = this.viewPieChartDescriptionStyle.getColors();
+        String strokeColor = this.viewPieChartDescriptionStyle.getStrokeColor();
+        String strokeWidth = this.viewPieChartDescriptionStyle.getStrokeWidth();
+        this.handleColors(pieChartStyleBuilder, colors);
+        this.handleStrokeColor(pieChartStyleBuilder, strokeColor);
+        this.handleStrokeWidth(pieChartStyleBuilder, strokeWidth);
+        this.handleLabelStyle(pieChartStyleBuilder, this.viewPieChartDescriptionStyle);
+        return pieChartStyleBuilder.build();
+    }
+
+    private void handleLabelStyle(Builder pieChartStyleBuilder, PieChartDescriptionStyle effectiveStyle) {
+        // @formatter:off
+        pieChartStyleBuilder.fontSize(effectiveStyle.getFontSize())
+            .bold(effectiveStyle.isBold())
+            .italic(effectiveStyle.isItalic())
+            .strikeThrough(effectiveStyle.isStrikeThrough())
+            .underline(effectiveStyle.isUnderline());
+        // @formatter:on
+    }
+
+    private void handleStrokeWidth(Builder pieChartStyleBuilder, String strokeWidth) {
+        if (strokeWidth != null && !strokeWidth.isBlank()) {
+            try {
+                int width = Integer.parseInt(strokeWidth);
+                pieChartStyleBuilder.strokeWidth(width);
+            } catch (NullPointerException | NumberFormatException e) {
+                // unexpected value.
+            }
+        }
+    }
+
+    private void handleStrokeColor(Builder pieChartStyleBuilder, String strokeColor) {
+        if (strokeColor != null && !strokeColor.isBlank()) {
+            pieChartStyleBuilder.strokeColor(strokeColor);
+        }
+    }
+
+    private void handleColors(Builder pieChartStyleBuilder, String colors) {
```

Comment:
This does not actually use the supplied `colors` string. It iterates on `colorsObjets` which has just been initialied to `List.of()`. Not sure what's the expected format for `colors` (I don't find it mentioned in the documentation).

Reply:
`colors` should have been named `colorsExpression`. An AQL expression is expected here, so this property should not be taken into account for the style preview.